keyboards/install.php

```diff
@@ -0,0 +1,315 @@
+<?php
+  declare(strict_types=1);
+
+  namespace Keyman\Site\com\keyman;
+
+  require __DIR__ . '/../_includes/autoload.php';
+
+  use Keyman\Site\com\keyman\templates\Head;
+  use Keyman\Site\com\keyman\templates\Menu;
+  use Keyman\Site\com\keyman\templates\Body;
+  use Keyman\Site\com\keyman\templates\Foot;
+  use Keyman\Site\com\keyman\templates\AppStore;
+  use Keyman\Site\com\keyman\templates\PlayStore;
+  use Keyman\Site\com\keyman\KeymanDownloadVersions;
+  use Keyman\Site\com\keyman\KeymanHosts;
+  use Keyman\Site\com\keyman\Util;
+
+  KeyboardInstallPage::render_keyboard_details(
+    isset($_REQUEST['id']) ? $_REQUEST['id'] : null,
+    isset($_REQUEST['tier']) ? $_REQUEST['tier'] : null,
+    isset($_REQUEST['bcp47']) ? $_REQUEST['bcp47'] : null
+  );
+
+  class KeyboardInstallPage
+  {
+    const BOOTSTRAP_SEPARATOR = '.';
+
+    // Properties for querying api.keyman.com
+    static private $id;
+    static private $tier;
+
+    // Properties to provide to apps in embedded download mode
+    static private $bcp47;
+
+    // Properties for querying keyboard downloads
+    static private $keyboard;  // from api.keyman.com/keyboard
+    static private $downloads; // from downloads.keyman.com/api/keyboard
+    static private $versions;  // from downloads.keyman.com/api/version
+    static private $title;
+    static private $error;
+
+    /**
+     * render_keyboard_details - display keyboard download boxes and details
+     * @param $id - keyboard ID
+     * @param string $tier - ['stable', 'alpha', or 'beta']
+     * @param bool $landingPage - when true, details won't display keyboard search box or title
+     * @param string $bcp47 - BCP 47 tag to pass as a hint to download links for apps to make connection
+     */
+    public static function render_keyboard_details($id, $tier, $bcp47) {
+      self::$id = $id;
+      self::$bcp47 = self::validate_bcp47($bcp47);
+      self::$tier = self::validate_tier($tier);
+
+      self::LoadData();
+      self::WriteTitle();
+      if (isset(self::$downloads->kmp)) {
+        self::WriteDownloadBoxes();
+      } else {
+        //TODO: self::WriteNoDownloadMessage();
```

Comment:
Intentional omission?

Reply:
The `TODO` indicates I need to do work here. It's on my list. Theoretically, a user should never get here if there is no .kmp as referring pages would not generate a link for a keyboard that has no .kmp (which would only be legacy web-only keyboards). Given that, it's low priority.